src/main/java/com/kobylynskyi/graphql/codegen/mapper/RequestResponseDefinitionToDataModelMapper.java

```diff
@@ -196,6 +197,12 @@ private static ProjectionParameterDefinition getTypeNameProjectionParameterDefin
         dataModel.put(METHOD_NAME, dataModelMapper.capitalizeMethodNameIfRestricted(mappingContext, operationDef.getName()));
         dataModel.put(RETURN_TYPE_NAME, javaType);
         dataModel.put(GENERATED_INFO, mappingContext.getGeneratedInformation());
+        // if getUseOptionalForNullableReturnTypes == true, append `?` for kotlin
```

Comment:
@jxnu-liguobin, I think it is better if you override `getLanguageType` inside `KotlinGraphQLTypeMapper` and append "?" there:

```java
    @Override
    public NamedDefinition getLanguageType(MappingContext mappingContext, String graphQLType, String name, String parentTypeName, boolean mandatory, boolean collection) {
        NamedDefinition languageType = GraphQLTypeMapper.super.getLanguageType(mappingContext, graphQLType, name, parentTypeName, mandatory, collection);
        if (mappingContext.getUseOptionalForNullableReturnTypes().equals(Boolean.TRUE)) {
            languageType.setJavaName(languageType.getJavaName() + "?");
        }
        return languageType;
    }
```

Then this will take affect in all underlying Mappers.

Reply:
yes,

Reply:
I will extract a`getResponseLanguageType` and rewrite it in different languages, What's your opinion?